code/modules/projectiles/guns/magic/staff.dm

```diff
@@ -54,3 +54,108 @@
 	max_charges = 10
 	recharge_rate = 2
 	no_den_usage = 1
+
+
+// Lord Staves
+
+/obj/item/weapon/gun/magic/staff/staffofrevenant
+	name = "staff of revenant"
+	desc = "A cursed artifact that starts off weak, but you can drain the souls of dead bodies in order to make it more powerful! Activate the staff in hand to see how many souls you have and, if you have enough, make your staff stronger."
+	fire_sound = "sound/magic/WandODeath.ogg"
+	ammo_type = /obj/item/ammo_casing/magic/staffofrevenant
+	icon_state = "staffofrevenant"
+	item_state = "staffofrevenant"
+	w_class = 4
+	max_charges = 1
+	recharge_rate = 10
+	no_den_usage = 1
+	var/revenant_level = 0
+	var/revenant_damage = 20 // doesn't actually change the damage, only used for the inhand activation
+	var/revenant_souls = 0
```

Comment:
I've had cases where mobs disappeared from the drained_mobs list if their body is destroyed in certain ways, hence why I use revenant_souls, so that the wizard doesn't get shafted by backward progress.